### PR TITLE
changed selects to be autocompletes in cat Items and items dialogs #495

### LINKED ISSUE #495: All list in the catalogue item should be autocompletes (catalogue items dialog)
Change select component to autocompletes in the catalogue items dialog

### PR DESCRIPTION
**Pull request by MatteoGuarnaccia5**
## Description

Changed the select inputs in catalogue items and items dialog to be autocompletes

## Testing instructions

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage

## Agile board tracking

closes #495

### CODE REVIEW

File: src/utils.tsx [hunk 1]
```diff
@@ -140,3 +140,10 @@ export const resetUniqueIdCounter = () => {
 export function sortDataList(data: any[], sortedValue: string) {
   return data.sort((a, b) => a[sortedValue].localeCompare(b[sortedValue]));
 }
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function addValueToFrontOfList(list: any[], newValue: any): any[] {
+  if (list[0] == 'None') return list;
+  list.unshift(newValue);
+  return list;
+}
```
[the remark this answers is not in the document]
Comment (MatteoGuarnaccia5) [reply src/utils.tsx:149]: Yeah I was going to remove it

File: src/items/itemDialog.component.test.tsx [hunk 2]
```diff
@@ -228,7 +220,7 @@ describe('ItemDialog', () => {
       createView();
 
       await modifyDetailsValues({
-        usageStatus: 'Used',
+        usageStatus: 'U{arrowdown}{arrowdown}{enter}',
```
Comment (joelvdavies) [inline src/items/itemDialog.component.test.tsx:223]:
```suggestion
        # In Use
        usageStatus: 'U{arrowdown}{arrowdown}{enter}',
```
Why does this one require two arrowdown's and the others only one? Is it worth putting a comment on what this is selecting? Same below.

Comment (MatteoGuarnaccia5) [reply src/items/itemDialog.component.test.tsx:223]: Alot of the tests have two arrowdowns as well - that selected `Used`. To be honest I don't know why I put some as just one arrowdown, I can make them all consistent? When it doesn't matter what the usage status is

Comment (joelvdavies) [reply src/items/itemDialog.component.test.tsx:223]: I think its worth doing it just for this file, I think they all for values that used to be `usageStatus: 'Used'`, but now it is selecting 'In Use' instead.

Comment (joelvdavies) [reply src/items/itemDialog.component.test.tsx:253]: @MatteoGuarnaccia5 Sorry could you remove the 2nd arrow down instead of adding them to the rest

**Comment by joshuadkitenge** [comment]
Can you also change the select components to autocomplete in the table column filters please

**Review by joelvdavies — DISMISSED** [review]
Worked well for me.

**Review by joelvdavies — APPROVED** [review]

**Pull request merged**